common/src/main/java/com/microsoft/identity/common/internal/providers/oauth2/IDToken.java

```diff
@@ -180,15 +182,15 @@ public IDToken(final String rawIdToken) {
         return Collections.unmodifiableMap(mTokenClaims);
     }
 
-    private Map<String, String> parseJWT(final String idToken) {
+    private Map<String, String> parseJWT(final String idToken) throws ServiceException {
         final String idTokenBody = extractJWTBody(idToken);
         final byte[] data = Base64.decode(idTokenBody, Base64.URL_SAFE);
 
         try {
             final String decodedBody = new String(data, Charset.forName(StringExtensions.ENCODING_UTF8));
             return JsonExtensions.extractJsonObjectIntoMap(decodedBody);
         } catch (final JSONException e) {
-            throw new RuntimeException(e);
+            throw new ServiceException("", ErrorStrings.INVALID_JWT, e);
```

Comment:
Could error code string be empty?

Reply:
keeping it empty for now, we need a diagnosibility push in the end.

Reply:
I'm still wondering if this is recoverable.... what are we communicating to the developer here?  What are they supposed to do to resolve this?